### PR TITLE
Rwa 78 workingdays

### PR DESCRIPTION
**Pull request by chrisgrimble**
Calculate the due for a task based on the working days.

If a request to create a task is made with a due date that is used, otherwise
the config contains the number of days for the the task. Calculate the due
date of the task based on working days.

For now working days are days that are not a weekend or not a holiday in
England or wales based on gov uk holiday dates. Holidays in other parts of
the country will need to be handled in a separate story.

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-78

**Does this PR introduce a breaking change?** (check one with "x")

```
[X] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/duedate/RealDateService.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.time.ZonedDateTime;
+
+@Component
+public class RealDateService implements DateService {
+    @Override
+    public ZonedDateTime now() {
+        return ZonedDateTime.now();
+    }
+}
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/duedate/RealDateService.java:13]: Suggestion, maybe the above does not need to be defined explicitly and we could use lambdas like the below example: 

`public class David {

    void test() {
        DateService dateService = ZonedDateTime::now;
        System.out.println(dateService);
    }
}`

File: src/test/java/uk/gov/hmcts/reform/waworkflowapi/duedate/DueDateServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        ZonedDateTime calculatedDueDate = underTest.calculateDueDate(
+            null,
+            new TaskToCreate(PROCESS_APPLICATION, TCW_GROUP, leadTimeDays)
+        );
+
+        assertThat(calculatedDueDate, is(now.plusDays(leadTimeDays)));
+    }
+
+    @Test
```
Comment (iguannago) [inline src/test/java/uk/gov/hmcts/reform/waworkflowapi/duedate/DueDateServiceTest.java:70]: maybe we could parametrise these tests and reuse the code. It looks like you repeat the very similar code for each scenario.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/duedate/RealDateService.java (hunk 1 above)
Comment (chrisgrimble) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/duedate/RealDateService.java]: Done

File: src/test/java/uk/gov/hmcts/reform/waworkflowapi/duedate/DueDateServiceTest.java (hunk 2 above)
Comment (chrisgrimble) [reply src/test/java/uk/gov/hmcts/reform/waworkflowapi/duedate/DueDateServiceTest.java:62]: I've removed a load of duplication from this test. If we were to use a parameterised tests can we still give each row a name which is printed out as part of the test. I think the names of the methods are important here e.g. calculateDueDateWhenStraddlesAWeekend and wonder if there is a way to keep that with a parameterised test?

Comment (iguannago) [reply src/test/java/uk/gov/hmcts/reform/waworkflowapi/duedate/DueDateServiceTest.java:62]: yes, it looks better now. I think it's possible,  I used to do with another parametrise library and don't know with junit 5.  maybe for the next time around:)

**Review by iguannago — APPROVED** [review]

**Pull request merged**